kernel_hardening_checker/checks.py

```diff
@@ -222,6 +222,9 @@ def add_kconfig_checks(l: List[ChecklistObjType], arch: str) -> None:
     l += [OR(KconfigCheck('self_protection', 'kspp', 'UBSAN_SANITIZE_ALL', 'y'),
              AND(ubsan_bounds_is_set,
                  VersionCheck((6, 9, 0))))] # UBSAN_SANITIZE_ALL was enabled by default in UBSAN in v6.9
+    l += [KconfigCheck('self_protection', 'kspp', 'LSM', '*landlock*')]
+    l += [KconfigCheck('self_protection', 'kspp', 'LSM', '*lockdown*')]
+    l += [KconfigCheck('self_protection', 'kspp', 'LSM', '*yama*')]
```

Comment:
Please add a missing `OR` check for major LSMs.

Reply:
like that?
```Python
    l += [KconfigCheck('security_policy', 'kspp', 'LSM', '*yama*'),
          OR(KconfigCheck('security_policy', 'a13xp0p0v', 'SECURITY_SELINUX', 'y'),
             KconfigCheck('security_policy', 'a13xp0p0v', 'SECURITY_APPARMOR', 'y'),
             KconfigCheck('security_policy', 'a13xp0p0v', 'SECURITY_SMACK', 'y'),
             KconfigCheck('security_policy', 'a13xp0p0v', 'SECURITY_TOMOYO', 'y'))]
    l += [KconfigCheck('security_policy', 'kspp', 'LSM', '*landlock*'),
          OR(KconfigCheck('security_policy', 'a13xp0p0v', 'SECURITY_SELINUX', 'y'),
             KconfigCheck('security_policy', 'a13xp0p0v', 'SECURITY_APPARMOR', 'y'),
             KconfigCheck('security_policy', 'a13xp0p0v', 'SECURITY_SMACK', 'y'),
             KconfigCheck('security_policy', 'a13xp0p0v', 'SECURITY_TOMOYO', 'y'))]
    l += [KconfigCheck('self_protection', 'kspp', 'LSM', '*lockdown*'),
          OR(KconfigCheck('security_policy', 'a13xp0p0v', 'SECURITY_SELINUX', 'y'),
             KconfigCheck('security_policy', 'a13xp0p0v', 'SECURITY_APPARMOR', 'y'),
             KconfigCheck('security_policy', 'a13xp0p0v', 'SECURITY_SMACK', 'y'),
             KconfigCheck('security_policy', 'a13xp0p0v', 'SECURITY_TOMOYO', 'y'))]
             ```

Reply:
No, I mean a new `*value*` OR-check for one of the major LSMs, like you've made for `landlock` of `yama`.

Reply:
Please see https://github.com/a13xp0p0v/kernel-hardening-checker/pull/167#discussion_r1895341839